src/datatypes.ts

```diff
@@ -193,71 +187,120 @@ export function serializeTime(value: Time): string {
     }
   }
 
+  if (value.timezoneHourOffset || value.timezoneHourOffset == 0) {
```

Comment:
If I'm understanding this correctly you basically want to check whether `value.timezoneHourOffset` is defined, but since 0 is implicitly cast to `false`, you also have to check for the value 0 (or rather, you're checking for values that can be cast to 0 by using `==` rather than `===`, so `false == 0` also matches).

To avoid potential footguns like that, when working in JavaScript it's generally a good habit to avoid implicit type casting:

```suggestion
  if (typeof value.timezoneHourOffset !== "undefined") {
```

or even checking explicitly whether it is of the correct type:

```suggestion
  if (typeof value.timezoneHourOffset === "number") {
```